### PR TITLE
[bitnami/contour] Add prometheusRule to contour

### PR DESCRIPTION
**Pull request by clayton-gonsalves**
### Description of the change

Add `PrometheusRule` to contour

### Benefits

Enables definitions of prometheus rules via helm. 

### Possible drawbacks

N/A

### Applicable issues

N/A

### Additional information

N/A

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/contour/values.yaml [hunk 1]
```diff
@@ -1289,20 +1289,18 @@ ingress:
...
-## @param metrics.serviceMonitor.interval Specify the scrape interval if not specified use default prometheus scrapeIntervall, the Prometheus default scrape interval is used.
-## @param metrics.serviceMonitor.metricRelabelings [array] Specify additional relabeling of metrics.
-## @param metrics.serviceMonitor.relabelings [array] Specify general relabeling.
-## @param metrics.serviceMonitor.honorLabels Specify honorLabels parameter to add the scrape endpoint
-## @param metrics.serviceMonitor.scrapeTimeout The timeout after which the scrape is ended
-## @param metrics.serviceMonitor.selector Specify honorLabels parameter to add the scrape endpoint
-## @param metrics.serviceMonitor.labels Extra labels for the ServiceMonitor
-##
 metrics:
+  ## Prometheus Operator service monitors
```
Comment (fevisera) [inline bitnami/contour/values.yaml:1293]: Could you follow the same format we use in other charts (e.g. Apache) for this kind of object? See
https://github.com/bitnami/charts/blob/main/bitnami/apache/values.yaml#L656

File: bitnami/contour/values.yaml [hunk 2]
```diff
@@ -1315,6 +1313,17 @@ metrics:
     selector: {}
     labels: {}
 
+  ## Prometheus Operator prometheusRules
+  ## @param metrics.prometheusRule.enabled Creates a Prometheus Operator prometheusRule (also requires `metrics..serviceMonitor.enabled` to be `true` and `metrics.prometheusRule.rules`)
+  ## @param metrics.prometheusRule.namespace Namespace for the prometheusRule Resource (defaults to the Release Namespace)
+  ## @param metrics.prometheusRule.additionalLabels Additional labels that can be used so prometheusRule will be discovered by Prometheus
+  ## @param metrics.prometheusRule.rules Prometheus Rule definitions
+  prometheusRule:
```
Comment (fevisera) [inline bitnami/contour/values.yaml:1321]: Could you follow the same format we use in other charts (e.g. Apache) for this kind of object? See
https://github.com/bitnami/charts/blob/main/bitnami/apache/values.yaml#L686

File: bitnami/contour/README.md [hunk 3]
```diff
@@ -399,18 +399,22 @@ helm uninstall my-release
...
+| `metrics.serviceMonitor.honorLabels`       | Specify honorLabels parameter to add the scrape endpoint                                                                                       | `false`                  |
+| `metrics.serviceMonitor.scrapeTimeout`     | The timeout after which the scrape is ended                                                                                                    | `""`                     |
+| `metrics.serviceMonitor.selector`          | Specify honorLabels parameter to add the scrape endpoint                                                                                       | `{}`                     |
+| `metrics.serviceMonitor.labels`            | Extra labels for the ServiceMonitor                                                                                                            | `{}`                     |
+| `metrics.prometheusRule.enabled`           | Creates a Prometheus Operator prometheusRule (also requires `metrics..serviceMonitor.enabled` to be `true` and `metrics.prometheusRule.rules`) | `false`                  |
```
Comment (fevisera) [inline bitnami/contour/README.md:414]: Correct the sentence in parentheses, maybe like this:
```
(also requires `metrics.serviceMonitor.enabled` and `metrics.prometheusRule.rules` to be `true`)
```

**Comment by fevisera** [comment]
Thanks so much for this contribution @clayton-gonsalves!
Please, check my comments when you have a chance.

**Comment by fevisera** [comment]
Also, please bump the chart version again.
Thanks.

File: bitnami/contour/Chart.yaml [hunk 4]
```diff
@@ -22,4 +22,4 @@ maintainers:
 name: contour
 sources:
   - https://github.com/bitnami/charts/tree/main/bitnami/contour
-version: 12.0.1
+version: 12.2.0
```
Comment (fevisera) [inline bitnami/contour/Chart.yaml:25]: Please ensure to update the chart version in accordance with the semver principles outlined in our [contributing guidelines](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#technical-requirements).
The current version is [12.0.4](https://github.com/bitnami/charts/blob/main/bitnami/contour/Chart.yaml#LL25C3-L25C16), so it should be incremented to 12.1.0.

Comment (clayton-gonsalves) [reply bitnami/contour/Chart.yaml]: done!

**Review by fevisera — APPROVED** [review]

**Pull request merged**